docs/README.md

```diff
@@ -0,0 +1,627 @@
+
+rand.choose((0 to 15).toSeq)
+
+rand.choose((0 to 15).toSeq)
+
+rand.choose((0 to 15).toSeq)
+
+rand.chooseN((0 to 15).toSeq, 4)
+
+rand.chooseN((0 to 15).toSeq, 4)
+```
+
+### JreVersionHelper
+
+The JreVersionHelper object provides helper methods to check the two most significant parts of the JRE version at runtime:
+
+```scala mdoc:compile-only
+import com.velocidi.apso.JreVersionHelper
+
+JreVersionHelper.jreVersion
+// res0: (Int, Int) = (1, 8)
+```
+
+### Logging
+
+The `Logging` and `StrictLogging` traits allows mixing in Log4j2 `Logger` objects. The difference between the two is that in the former the `Logger` object is initialized lazily, while in the latter it is initialized strictly:
+
+```scala mdoc:compile-only
+import com.velocidi.apso.Logging
+
+class A extends Logging {}
+
+val a = new A
+
+a.log.info("test")
+```
+
+### ProgressBar
+
+The `ProgressBar` represents a widget to print a dynamic progress bar in a console.
+
+```scala mdoc:compile-only
+import com.velocidi.apso.ProgressBar
+
+val progress = ProgressBar(100)
+
+progress.tick(1)
+// 1% [>                                                     ] / [ 0.19 ] ops/s	
+
+progress.tick(2)
+// 3% [=>                                                    ] - [ 0.15 ] ops/s
+
+progress.tick(1)
+// 4% [==>                                                   ] \ [ 0.12 ] ops/s
+
+progress.tick(10)
+// 14% [=======>                                              ] | [ 0.31 ] ops/s
+
+progress.tick(20)
+// 34% [==================>                                   ] / [ 0.46 ] ops/s
+
+progress.tick(30)
+// 64% [=================================>                    ] - [ 0.77 ] ops/s
+```
+
+### Reflect
+
+The `Reflect` object contains helpers for reflection-related tasks, namely to create an instance of a given class given its fully qualified name and also to access singleton objects:
+
+```scala
+scala> import com.velocidi.apso.Reflect
+import com.velocidi.apso.Reflect
+
+scala> import com.velocidi.apso.collection._
+import com.velocidi.apso.collection._
+
+scala> Reflect.newInstance[HMap[Nothing]]("com.velocidi.apso.collection.HMap")
+res0: com.velocidi.apso.collection.HMap[Nothing] = HMap()
+
+scala> Reflect.companion[Reflect.type]("com.velocidi.apso.Reflect")
+res1: com.velocidi.apso.Reflect.type = com.velocidi.apso.Reflect$@3b1dbca
+```
+
+### Retry
+
+The `Retry` object provides a method to retry methods or `Future`s a given number of times until they succeed or the specified maximum number of retries is reached:
+
+```scala mdoc:reset
+import scala.concurrent._
+import scala.concurrent.duration._
+import scala.concurrent.ExecutionContext.Implicits.global
+
+import com.velocidi.apso.Retry
+
+import java.util.concurrent.atomic.AtomicInteger
+
+val a = new AtomicInteger()
+
+def f: Future[Int] = {
+  Future {
+    val value = a.getAndAdd(1)
+    if (value > 5)
+      value
+    else {
+      throw new Exception()
+    }
+  }
+}
+
+Await.result(Retry.retryFuture(10)(f), 1.second)
```

Comment:
Since we're sure that this completes, we can be more lenient and await indefinitely.

```suggestion
Await.result(Retry.retryFuture(10)(f), Duration.Inf)
```